app/models/user.rb

```diff
@@ -5,6 +5,59 @@ class User < ApplicationRecord
   include Searchable
   include Storext.model
 
+  # NOTE: @citizen428 This is temporary code during profile migration and will
+  # be removed.
+  # rubocop:disable Metrics/BlockLength
+  concerning :ProfileMigration do
+    included do
+      PROFIE_FIELDS =
+        (Profiles::ExtractData::DIRECT_ATTRIBUTES + Profiles::ExtractData::MAPPED_ATTRIBUTES.values).freeze
+
+      # NOTE: this allows us to skip the profile create callback in a thread-safe way
+      attr_accessor :_skip_creating_profile
```

Comment:
Tired: using `skip_callback` which modifies global state and can lead to hard to track down bugs since it's not thread-safe.
Wired: using an instance-level `attr_accessor` so we don't add more flakiness to our tests.

Reply:
ahha neat trick! though It's odd to have the underscore. Is it only going to be used in the tests?

Reply:
Yes, it's only supposed to be used in tests, the underscore is meant to be a hint to that. It's not a completely uncommon practice btw, I've seen the same pattern in other codebases.

Reply:
TIL! I'm going to guess not many people know about this. Might be worth a little blog post :) 

Should we put a comment here that the purpose of this is for tests? I had to go back through the PR conversation to figure out why it was here. I could see someone else having the same question

Reply:
@mstruve Yes, I already made a note to write up a post. I'll update the comment now.